community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/NeoStoreUtil.java

```diff
@@ -49,47 +46,57 @@
 {
     private final Map<Position,Long> values = new HashMap<>();
 
-    public static void main( String[] args )
+    public static boolean neoStoreExists( PageCache pageCache, File storeDir )
     {
-        if ( args.length < 1 )
+        try ( PagedFile file = pageCache.map( neoStoreFile( storeDir, StoreFileType.STORE ), pageCache.pageSize() ) )
         {
-            System.err.println( "Supply one argument which is the store directory of a neo4j graph database" );
-            System.exit( 1 );
+            if ( file.getLastPageId() == -1 )
+            {
+                return false;
+            }
+        }
+        catch ( IOException e )
+        {
+            return false;
         }
-        System.out.println( new NeoStoreUtil( new File( args[0] ) ) );
-    }
 
-    public static boolean neoStoreExists( FileSystemAbstraction fs, File storeDir )
-    {
-        return fs.fileExists( neoStoreFile( storeDir, StoreFileType.STORE ) );
+        return true;
     }
 
-    public NeoStoreUtil( File storeDir )
-    {
-        this( storeDir, new DefaultFileSystemAbstraction() );
-    }
-
-    public NeoStoreUtil( File storeDir, FileSystemAbstraction fs )
+    public NeoStoreUtil( File storeDir, PageCache pageCache )
```

Comment:
Why not having a static method `NeoStoreUtil.parse()` for parsing the NeoStore which will return an immutable object containing the data?